editor/scene_tree_dock.cpp

```diff
@@ -2702,13 +2707,16 @@ void SceneTreeDock::_new_scene_from(String p_file) {
 	}
 }
 
-void SceneTreeDock::_set_node_owner_recursive(Node *p_node, Node *p_owner) {
-	if (!p_node->get_owner()) {
-		p_node->set_owner(p_owner);
+void SceneTreeDock::_set_node_owner_recursive(Node *p_node, Node *p_owner, HashMap<const Node *, Node *> &inverse_duplimap) {
+	if (inverse_duplimap.has(p_node)) {
+		Node *original = inverse_duplimap.get(p_node);
```

Comment:
```suggestion
	HashMap<const Node *, Node *>::Iterator E = inverse_duplimap.find(p_node);
	if (E) {
		Node *original = E.value;
```
Avoids double lookup.